mmdeploy/codebase/mmocr/models/text_recognition/abinet_language_decoder.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmocr.models.textrecog.decoders.ABILanguageDecoder'
+    '._get_length',
+    backend='default')
+def abi_language_decoder__get_length(ctx, self, logit, dim=-1):
+    """Rewrite `_get_length` of ABILanguageDecoder for default backend. Greedy
+    decoder to obtain length from logit. Returns the first location of padding
+    index or the length of the entire tensor otherwise.
+
+    Rewrite this function to:
+    1. use torch.max to replace original torch.any and torch.cumsum.
+    2. modify the data type of out.new_tensor().
+    """
+    # out as a boolean vector indicating the existence of end token(s)
+    out = (logit.argmax(dim=-1) == self.pad_idx)
+    out = out.to(torch.float32)
+    abn, out = torch.max(out, dim)
```

Comment:
As the origin function has comment `Get the first index of end token` here. I don't think it is ok to replace `cumsum` with `max`.